### ISSUE TITLE
Checkmate not detected in a caparandom game

### ISSUE THREAD

**Issue by rchastain** [cutechess/cutechess#755]
Hello!

In a _caparandom_ tournament, **cutechess-cli** says that my engine played an illegal move. When I load the PGN of the game in **xboard**, it appears that in fact the engine was checkmated.

My version of **cutechess-cli**:

```
cutechess-cli 1.3.0-beta4
Using Qt version 5.15.2
Running on Mageia 8/x86_64
```

How I ran **cutechess-cli**:

```bash
# tournoi-capablanca.sh
cc=/home/roland/Documents/echecs/sources/cutechess/130/beta4/build/cutechess-cli

rm -f *.err
rm -f *.log
rm -f *.pgn

$cc \
-rounds 100 \
-variant caparandom \
-repeat \
-engine name="Durandal 0.1.3" cmd=durandal64 dir=. proto=uci \
-engine conf="Durandal 0.1.0" \
-engine conf="Durandal 0.1.1" \
-engine conf="Durandal 0.1.2" \
-engine name="Durandal 0.1.1 [random]" cmd=random64 dir=. proto=uci \
-pgnout tournoi-capablanca.pgn \
-each tc=40/20 \
-wait 2000 2>> tournoi-capablanca.err | tee tournoi-capablanca.log
```

The PGN produced by **cutechess-cli**:

```
[Event "?"]
[Site "?"]
[Date "2023.02.02"]
[Round "220"]
[White "Durandal 0.1.0"]
[Black "Durandal 0.1.3"]
[Result "1-0"]
[FEN "rknbqrbacn/pppppppppp/10/10/10/10/PPPPPPPPPP/RKNBQRBACN w KQkq - 0 1"]
[GameDuration "00:00:02"]
[GameEndTime "2023-02-02T17:54:32.440 CET"]
[GameStartTime "2023-02-02T17:54:29.861 CET"]
[PlyCount "47"]
[SetUp "1"]
[Termination "illegal move"]
[TimeControl "40/20"]
[Variant "caparandom"]

1. g3 g5 2. e3 f5 3. d4 Bc4 4. Ne2 Ch6 5. c3 e6 6. i3 Bf6 7. a3 Qh5 8. Nf4 Qi4
9. h3 Bxf1 10. hxi4 gxf4 11. Qxf1 fxe3 12. fxe3 Bxd4 13. exd4 f4 14. gxf4 Ag6+
15. Bc2 Axc2+ 16. Kxc2 Rg8 17. Be3 Rg1 18. Bxg1 Cg4 19. Re1 Ne7 20. f5 Nxf5
21. Re4 Cg5 22. Re5 Nxd4+ 23. cxd4 Cg4
24. Qf8+ {Black makes an illegal move: g4e5} 1-0
```

The final position, according to **xboard**:

![bug230202](https://user-images.githubusercontent.com/13957204/216422770-b1511c15-abf3-4918-827f-ca7940440129.png)

I see that there are many issues opened. I don't want to bore you. Thank you for your great work on Cutechess.

Regards.

Roland

**Comment by alwey** [comment]
This problem is solved by #617 which fixes generalized (Chess960) castling.

References: #616 #637, #754.

**Comment by rchastain** [comment]
Great! Thank you.
